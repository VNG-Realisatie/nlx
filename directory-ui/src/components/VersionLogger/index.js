// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//

import { useState, useEffect } from 'react'
import PropTypes from 'prop-types'

const VersionLogger = ({ logger }) => {
  const [versionTag, setVersionTag] = useState(null)

  useEffect(
    () => {
      const load = async () => {
        try {
          const result = await fetch('/version.json')
          const { tag } = await result.json()
          setVersionTag(tag)
        } catch (e) {}
      }
      load()
    },
    [], // prevent inifinite rerenders
  )

  if (versionTag) {
    logger(versionTag)
  }
  return null
}

VersionLogger.propTypes = {
  logger: PropTypes.func,
}

VersionLogger.defaultProps = {
  // eslint-disable-next-line no-console
  logger: console.log,
}

export default VersionLogger
