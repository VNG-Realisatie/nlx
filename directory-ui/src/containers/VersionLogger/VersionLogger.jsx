// Copyright Â© VNG Realisatie 2019
// Licensed under the EUPL

import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types';
 
const VersionLogger = ({logger}) => {
    const [versionTag, setVersionTag] = useState(null)
    
    useEffect(
        () => {
        const load = async () => {
            const result = await fetch('/version.json')
            const {tag} = await result.json()
            setVersionTag(tag)
        }
        load()
    },
    [], // prevent inifinite rerenders
    )
    
    if (!!versionTag) {
        logger(versionTag)
    }
    return null
}

VersionLogger.propTypes = {
    logger: PropTypes.func,
}

VersionLogger.defaultProps = {
    logger: console.log,
}

export default VersionLogger
