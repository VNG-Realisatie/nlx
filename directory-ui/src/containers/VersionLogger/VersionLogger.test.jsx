import React from 'react';
import {shallow, mount} from 'enzyme';
import VersionLogger from './VersionLogger';
import { act, create } from "react-test-renderer";

describe('VersionLogger', () => {
    beforeEach(() => {
        fetch.resetMocks()
        fetch.mockResponseOnce(JSON.stringify({ tag: 'test' }))
    })

    describe('effect', () => {
        it('should call logger when rendered', (done) => {
            const logger = jest.fn()
            act(() => {
                // the async call in useEffect will cause a warning. this will pro
                create(<VersionLogger logger={logger}/>)
            })
            expect(fetch).toHaveBeenCalledTimes(1)
            
            // workaround for async result of fetch
            setTimeout(() => {
                expect(logger).toHaveBeenCalledWith('test')
                done()
            })
        })
    })

    describe('snapshot', () => {
        it('should should match snapshot', () => {
            const logger = jest.fn()

            expect(shallow(<VersionLogger logger={logger}/>)).toMatchSnapshot()
        });
    });
});