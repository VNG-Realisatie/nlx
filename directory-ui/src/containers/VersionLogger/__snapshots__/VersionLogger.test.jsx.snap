// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionLogger snapshot should should match snapshot 1`] = `""`;
